### PR TITLE
parser, ast: add `SET ROLE` support

### PR DESCRIPTION
**Pull request by imtbkcat**
### What problem does this PR solve? 
`SET ROLE` is an important part of RBAC function. Many other RBAC grammar rely on `SET ROLE`. The detail document for `SET ROLE` is here : https://dev.mysql.com/doc/refman/8.0/en/set-role.html

### What is changed and how it works?
Adding `RoleIdentity` and `SetRoleStmt` for `ast` package, which contain information about those role is going to be active.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - Increased code complexity

### CODE REVIEW

File: go.mod1 [hunk 1]
```diff
@@ -11,6 +11,7 @@ require (
 	github.com/pingcap/errors v0.11.0
 	github.com/pingcap/tidb v0.0.0-20190218065808-69472bd1a6e9
 	github.com/pingcap/tipb v0.0.0-20190107072121-abbec73437b7
+	github.com/pkg/errors v0.8.1
```
Comment (kennytm) [inline go.mod1]: Shouldn't need to change go.mod

Comment (imtbkcat) [reply go.mod1]: fixed

File: auth/auth.go [hunk 2]
```diff
@@ -67,6 +62,26 @@ func (user *UserIdentity) AuthIdentityString() string {
...
+		ctx.WritePlain("@")
+		ctx.WriteName(role.Hostname)
+	}
+	return nil
+}
+
+// String converts UserIdentity to the format user@host.
+func (role *RoleIdentity) String() string {
+	// TODO: Escape username and hostname.
+	return fmt.Sprintf("%s@%s", role.Username, role.Hostname)
```
Comment (tiancaiamao) [inline auth/auth.go]: `%s`@`%s` ?

**Comment by tiancaiamao** [comment]
LGTM

**Comment by imtbkcat** [comment]
tidb pr: https://github.com/pingcap/tidb/pull/9581

**Comment by tiancaiamao** [comment]
PTAL @jackysp  @lysu @kennytm

File: ast/misc.go [hunk 3]
```diff
@@ -721,6 +722,62 @@ func (n *SetPwdStmt) Accept(v Visitor) (Node, bool) {
...
+	case SetRoleDefault:
+		ctx.WriteKeyWord(" DEFAULT")
+	case SetRoleNone:
+		ctx.WriteKeyWord(" NONE")
+	case SetRoleAll:
+		ctx.WriteKeyWord(" ALL")
+	case SetRoleAllExcept:
+		ctx.WriteKeyWord(" ALL EXCEPT")
+	}
+	if len(n.RoleList) > 0 {
```
Comment (kennytm) [inline ast/misc.go]: This check is unnecessary.

File: go.mod1 [hunk 4]
```diff
@@ -11,6 +11,7 @@ require (
 	github.com/pingcap/errors v0.11.0
 	github.com/pingcap/tidb v0.0.0-20190218065808-69472bd1a6e9
 	github.com/pingcap/tipb v0.0.0-20190107072121-abbec73437b7
+	github.com/shirou/gopsutil v2.18.12+incompatible // indirect
```
Comment (kennytm) [inline go.mod1]: Where is this used? 🤔

**Comment by imtbkcat** [comment]
fixed @kennytm

**Pull request merged**